tests/tasks/test_notebook.py

```diff
@@ -150,25 +151,61 @@ def test_notebook_conversion_stores_as_unicode(tmp_directory, monkeypatch):
 
 
 @pytest.mark.parametrize(
-    "name, out_dir",
+    "name, out_dir, executor",
     [
-        ["sample.py", "."],
-        ["sample.R", "."],
-        ["sample.ipynb", "."],
+        ["sample.py", ".", "papermill"],
+        ["sample.R", ".", "papermill"],
+        ["sample.ipynb", ".", "papermill"],
         # check still works even if the folder does not exit yet
-        ["sample.ipynb", "missing_folder"],
+        ["sample.ipynb", "missing_folder", "papermill"],
+        # For Ploomber executor
+        ["sample.py", ".", "ploomber-engine"],
+        ["sample.R", ".", "ploomber-engine"],
+        ["sample.ipynb", ".", "ploomber-engine"],
+        # check still works even if the folder does not exit yet
+        ["sample.ipynb", "missing_folder", "ploomber-engine"],
+        ["sample.py", ".", "some-other-executor"],
+        ["sample.R", ".", "some-other-executor"],
+        ["sample.ipynb", ".", "some-other-executor"],
     ],
 )
-def test_execute_sample_nb(name, out_dir, tmp_sample_tasks):
+def test_execute_sample_nb(name, out_dir, executor, tmp_sample_tasks):
```

Comment:
This test is now testing many things at once, we should break it into separate test cases:

1. test changing the `executor`: None, papermill, and ploomber-engine
2. test the "Invalid executor error"
3. test the error when trying to execute  an R notebook

Reply:
I don't see the applied changes to the test. unit tests shouldn't have conditional logic, and you have an if statement there. Instead, they should be separate tests: one to run valid notebooks, one to check the error if the executor isn't valid, and another one to check the error when attempting to running an R notebook